eng/docgeneration/Generate-DocIndex.ps1

```diff
@@ -12,153 +27,95 @@ Write-Verbose "Initializing Default DocFx Site..."
-        # or management.
-        $PomPath = Join-Path -Path $Pkg -ChildPath "pom.xml"
-
-        # no pom file = nothing to process
-        if (Test-Path -path $PomPath) 
-        {
-            $xml = New-Object xml
-            $xml.Load($PomPath)
-
-            # Get the artifactId from the POM
-            $artifactId = $xml.project.artifactId
-
-            $parent = $xml.project.parent.relativePath
-
-            # If this is an artifact that isn't shipping then just
-            # move on to the next one
-            if ($ArtifactsToSkip -contains $artifactId)
-            {
-                Write-Output "skipping $artifactId"
-                continue
-            }
-
-            # If the parent is null or empty then the pom isn't directly including 
-            # one of the pom.[client|data|management].xml and needs to be specially
-            # handled
-            if (("" -eq $parent) -or ($null -eq $parent))
-            {
-                # Cosmos has a root pom which includes pom.client.xml that won't
-                # be detected by this logic. It's easier to deal with specially
-                # than it is to try and climb the pom chain here.
-                if ($Dir.BaseName -eq 'cosmos') 
-                {
-                    $clientArr += $artifactId
-                } 
-                else 
-                {
-                    Write-Host "*snowflake* Pom $PomPath, has a null or empty relative path."
-                }
-            } 
-            else 
-            {
-                if (($parent.IndexOf('azure-client-sdk-parent') -ne -1) -or ($parent.IndexOf('azure-data-sdk-parent') -ne -1))
-                {
-                    $clientArr += $artifactId
-                } 
-                else 
-                {
-                    $mgmtArr += $artifactId
-                }
-            }
-        }
+    else {
+        # Next page call
+        $resp = Invoke-RestMethod -Method Get -Uri "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=$langRegex&marker=$pageToken"
     }
-    # Only create this if there's something to create
-    #if (($clientArr.Count -gt 0) -or ($mgmtArr.Count -gt 0))
-    if ($clientArr.Count -gt 0)
-    {
-    New-Item -Path $YmlPath -Name "$($Dir.Name).md" -Force
-    Add-Content -Path "$($YmlPath)/toc.yml" -Value "- name: $($mappedDir)`r`n  href: $($Dir.Name).md"
-    # loop through the arrays and add the appropriate artifacts under the appropriate headings
-    if ($clientArr.Count -gt 0)
-    {
-        Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "# Client Libraries"
-        foreach($lib in $clientArr) 
-        {
-            Write-Host "Write $($lib) to $($Dir.Name).md"
-            Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "#### $lib"
+    # Storage returns some weired encoded string at the begining of response which needs to cutoff before use.
+    $rawConent = $resp.Substring(3)
```

Comment:
This seems a little odd, I suspect it is some encoding stuff and we should figure out how to correctly decode instead of blindly stripping 3 chars.

Reply:
Yes, I remember Java Storage met something similar when developing. Will consult and fix it if possible.